src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -6977,13 +6977,24 @@ export class StudyViewPageStore
                 chartUserSettings.hugoGeneSymbol &&
                 chartUserSettings.profileType
             ) {
+                const molecularProfileOptionByTypeMap = _.keyBy(
```

Comment:
since you are only attempting to find one item here, i think it would be as fast and much more clear just to use _.find() on the options array.